smac/tae/execute_func.py

```diff
@@ -172,11 +177,37 @@ def run(self, config, instance=None,
             runtime = time.time() - start_time
             additional_run_info = {}
 
+        # check serializability of results - if not, then attempt to convert it
```

Comment:
I don't think we want to convert data without communicating this to the user. Also, I think the user should convert the data herself to know about the consequences. Could you please remove this functionality?

Also, please only try to serialize it only to check if it is possible. The current implementation creates a json, which is than later on formatted as a json.

Finally, the current implementation wouldn't note which value cannot be serialized. Could you please make sure that this is part of the error message?